### PR TITLE
Add server deduplication by name

### PR DESCRIPTION
**Pull request by geeeezmo**

### CODE REVIEW

File: commands/bf1serverlist.js [hunk 1]
```diff
@@ -54,23 +54,45 @@ function parse(body) {
...
-					try {
-						const parsed = parse(body);
-						return resolve(parsed);
-					} catch (e) {
-						reject(e);
-					}
-				});
-			})
+			.then(doParse)
+			.then(doDedupe)
```
Comment (xorz57) [inline commands/bf1serverlist.js]: Nice! It looks very clean!

File: commands/bf1serverlist.js [hunk 2]
```diff
@@ -54,23 +54,45 @@ function parse(body) {
...
+function doParse(body) {
+	return new Promise((resolve, reject) => {
+		try {
+			const parsed = parse(body);
+			return resolve(parsed);
+		} catch (e) {
+			reject(e);
+		}
+	});
+}
```
Comment (xorz57) [inline commands/bf1serverlist.js]: Do we need to write anything to console when an exception happens?

File: commands/bf1serverlist.js [hunk 3]
```diff
@@ -54,23 +54,45 @@ function parse(body) {
...
+
+function doDedupe(servers, prop = 'name') {
+	return new Promise((resolve, reject) => {
+		try {
+			let uniqueNames = servers.reduce((acc, server) => {
+				if (!acc.includes(server[prop])) {
+					acc.push(server[prop]);
+				}
+				return acc;
+			}, []);
```
Comment (xorz57) [inline commands/bf1serverlist.js]: I guess you did something like [this](https://medium.com/dailyjs/how-to-remove-array-duplicates-in-es6-5daa8789641c)

**Comment by xorz57** [comment]
@all-contributors please add @geeeezmo for Maintenance.

**Comment by xorz57** [comment]
Just found a bug. If you try to make some duplicates your self, they still exist!

image

**Comment by xorz57** [comment]
I read some articles related to your solution and I came across this one https://dev.to/vuevixens/removing-duplicates-in-an-array-of-objects-in-js-with-sets-3fep
```node
function doDedupe(servers) {
	return new Promise((resolve, reject) => {
		try {
			const uniqueServers = Array.from(new Set(servers.map(server => server.name)))
			.map(name => {
			  return servers.find(server => server.name === name)
			})
			return resolve(uniqueServers);
		} catch (e) {
			reject(e);
		}
	});
}
```

**Comment by geeeezmo** [comment]
> @geeeezmo I read some articles related to your solution and I came across [this one](https://dev.to/vuevixens/removing-duplicates-in-an-array-of-objects-in-js-with-sets-3fep)
> 
> ```js
> function doDedupe(servers) {
> 	return new Promise((resolve, reject) => {
> 		try {
> 			const uniqueServers = Array.from(new Set(servers.map(server => server.name)))
> 			.map(name => {
> 			  return servers.find(server => server.name === name)
> 			})
> 			return resolve(uniqueServers);
> 		} catch (e) {
> 			reject(e);
> 		}
> 	});
> }
> ```

That is why I don't like arrays and prefer objects.
Too much kerfuffle.

**Comment by geeeezmo** [comment]
Checked that solution of yours and it works.

**Comment by xorz57** [comment]
@geeeezmo Well, if you also approve it, then we are OK!

**Pull request merged**